### ISSUE TITLE
Enhanced Observability

### ISSUE THREAD

**Issue by jonatan-ivanov** [spring-projects/spring-boot#25476]
This is a placeholder issue to group smaller tasks together to provide better Observability support for Spring Boot users.

# Info enhancements
- [ ] Add details about the JVM, e.g.: name, vendor, version
- [ ] Add details about the OS, e.g.: arch, name, version
- [ ] Add the list of `activeProfiles`
- [ ] Add details about the available resources e.g.: cpu, memory
- [ ] Add details about the used GC
- [ ] Add details about certain "user" properties: timezone, country, language, user dir
- [ ] Add network-related details: host, ip
- [ ] Add arbitrary records from the manifest file, see: https://github.com/spring-projects/spring-boot/issues/22924
- [ ] Add details about the certificate chain in case of HTTPS (alias, subject, issuer, version, serial, valid/expire dates, etc.)
- [ ] Miscellaneous items: current time, start time, uptime
- [ ] Enhance [`StartupInfoLogger`](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/StartupInfoLogger.java) with some of these details

See:
- An all-in-one implementation example for some of the features above:[RuntimeInfoContributor](https://github.com/jonatan-ivanov/teahouse/blob/master/core/src/main/java/org/example/teahouse/core/actuator/info/RuntimeInfoContributor.java)
- An example [json output](https://gist.github.com/jonatan-ivanov/1ec0d15c706ab439b38523f82340d41e#file-info-json)

# Health enhancements
- [ ] Add OOB health support for Feign clients: this should go to Spring Cloud OpenFeign, Spring Boot might provide support to handle the Health response (see: [this solution](https://github.com/jonatan-ivanov/teahouse/tree/master/core/src/main/java/org/example/teahouse/core/actuator/health))
- [ ] Metrics/tracing for health indicators

# Spring Security enhancements
- [ ] `/whoami` endpoint: show principal and granted authorities  

The list is still in progress and very likely to evolve in the near future.

**Comment by dsyer** [comment]
The `activeProfiles` is already exposed in the `/anv` endpoint.

**Comment by jonatan-ivanov** [comment]
Good point, today I was implementing it and I realized that if I don't set the active profiles property but set `spring.profiles.default`, then `environment.getActiveProfiles()` will be empty and the profle I set for default will be used. I think on the info endpoint it would be more useful to return the "effectively used profiles" instead of the value of `environment.getActiveProfiles()`. Something like this I guess:

```java
String[] getEffectiveProfiles() {
    return ObjectUtils.isEmpty(environment.getActiveProfiles()) ? environment.getDefaultProfiles() : environment.getActiveProfiles();
}
```

**Comment by dsyer** [comment]
Uptime is already a metric as well - I guess it really belongs there because you don't expect info to change over time.

**Comment by jonatan-ivanov** [comment]
I think using data that can change over time should be ok on the info endpoint, e.g.: current time (to detect clock skew issues), uptime (to see if the app was restarted or not), even [the cpu count can change over time](https://docs.oracle.com/javase/8/docs/api/java/lang/Runtime.html#availableProcessors--) (because of cgroups).

I think the same about data that we have metrics for, e.g.: [jvm version info](https://github.com/micrometer-metrics/micrometer/blob/main/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmInfoMetrics.java). I think they can be used in different use-cases and the user should decide if they want one or the other or both.
